server/routes/utils/reportHelper.ts

```diff
@@ -14,89 +14,136 @@
  */
 import puppeteer from 'puppeteer';
 import { v1 as uuidv1 } from 'uuid';
+import { LegacyObjectToConfigAdapter } from '../../../../../src/core/server/legacy';
 import { FORMAT } from './constants';
 
 export const generatePNG = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
-    await page.setDefaultNavigationTimeout(0);
-    await page.goto(url, { waitUntil: 'networkidle0' });
-
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
-
-    // TODO: this element is for Dashboard page, need to think about addition params to select html element with source(Visualization, Dashboard)
-
-    const timeCreated = new Date().toISOString();
-    const fileName = getFileName(itemName, timeCreated) + '.' + FORMAT.png;
-
-    const buffer = await page.screenshot({
-      fullPage: true,
-    });
-
-    //TODO: Add header and footer, phase 2
-
-    await browser.close();
-    return { timeCreated, dataUrl: buffer.toString('base64'), fileName };
+    return renderReport(
+      'image/png',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
   } catch (error) {
     throw error;
   }
 };
 
 export const generatePDF = async (
   url: string,
+  source: string,
   itemName: string,
   windowWidth: number,
-  windowHeight: number
+  windowHeight: number,
+  header: string,
+  footer: string
 ): Promise<{ timeCreated: string; dataUrl: string; fileName: string }> => {
   try {
-    const browser = await puppeteer.launch({
-      headless: true,
-    });
-    const page = await browser.newPage();
+    return renderReport(
+      'application/pdf',
+      url,
+      source,
+      itemName,
+      windowWidth,
+      windowHeight,
+      header,
+      footer
+    );
+  } catch (error) {
+    throw error;
+  }
+};
 
-    await page.setDefaultNavigationTimeout(0);
-    await page.setViewport({
-      width: windowWidth,
-      height: windowHeight,
-    });
+async function renderReport(
+  format: string,
+  url: string,
+  source: string,
+  itemName: string,
+  windowWidth: number,
+  windowHeight: number,
+  header: string,
+  footer: string
+) {
+  if (source === 'Dashboard') {
+    windowWidth = 1440;
```

Comment:
Not sure why dimension is different for Dashboard vs Visualization?

Reply:
Dashboards are larger- when the same smaller dimensions are applied, the dashboard is awkwardly bunched together/cut off. When the larger dimensions are applied to both, the Visualizations reports have too much empty space.